libraries/android/demo/build.gradle

```diff
@@ -1,37 +1,37 @@
 plugins {
     id 'com.android.application'
     id 'org.jetbrains.kotlin.android'
+    id 'com.facebook.react'
 }
 
 repositories {
     mavenCentral()
     google()
 }
 
+react {
+    root = file("../../../")
+    entryFile = file("../../../index.tsx")
+    reactNativeDir = file("../../../node_modules/react-native")
+    codegenDir = file("../../../node_modules/react-native-codegen")
+}
```

Comment:
If I'm not mistaken, this means that we'll also need to apply `com.facebook.react` to `WCAndroid` and introduce `react {` extension to configure it. 

I'm worried if applying `com.facebook.react` is safe for `WCAndroid` - it feels fragile, but I don't have arguments to support it.

Reply:
Since we'll package the `library` module into an artifact, this shouldn't be the case. However, I have doubts that it'll work out of the box and we'll likely need to do some of what the React Native Gradle Plugin is doing ourselves. For example, I am hoping that the plugin substitutes the `react-native-*` library dependencies with the artifacts from `mavenCentral`, as it suggests in its document. However, if it doesn't do that, since we won't have access to the `node_modules` folder from WCAndroid, we'll have to make sure we point to an artifact for these dependencies.

In short, I don't know if this setup will work for WCAndroid when we use binary dependencies. However, I don't think we'll ever want to apply `com.facebook.react` plugin to WCAndroid. Whatever issues come up, we'll have to deal with them some other way.

We should consider this PR just a step for you to be able to work on the project. Hope that makes sense.

Reply:
>  However, I don't think we'll ever want to apply com.facebook.react plugin to WCAndroid. Whatever issues come up, we'll have to deal with them some other way.

Gotcha, I think it's a good approach 👍 .

> We should consider this PR just a step for you to be able to work on the project. Hope that makes sense.

My biggest concern about this PR was that it'll break the included builds configuration with WCAndroid and, therefore, block our further progress. But, I managed to make it work by:
- Updating Gradle Wrapper to 7.5.1 [722e761](https://github.com/woocommerce/woocommerce-android/pull/9225/commits/722e761349d9cb362fda8a2f727c19cca8aaebbb)
- Updating AGP to 7.4.2 [1d1f720](https://github.com/woocommerce/woocommerce-android/pull/9225/commits/1d1f720463e0d7b187461e623216830a1823d859)
- Updating Aztec to v1.6.4 (previous version caused some DEX merging failing) [8fb35c0](https://github.com/woocommerce/woocommerce-android/pull/9225/commits/8fb35c0bbeb49fbaef85e7f0681cd281d4c8df62)
- Substituting React Native invalid dependencies with correct ones in `dependencySubstitiution` block [d976f25](https://github.com/woocommerce/woocommerce-android/pull/9225/commits/d976f2547385e083154d77fd6e7ba89014231612)

So, as after those changes we can still include `WooCommerce-Shared` to `WCAndroid`, I have no concerns with merging this PR 👍 .